modules/core/src/main/java/org/apache/ignite/cache/QueryEntity.java

```diff
@@ -669,6 +687,74 @@ public QueryEntity addQueryField(String fullName, String type, String alias) {
         return this;
     }
 
+    /**
+     * Returns INLINE_SIZE for PK index.
+     *
+     * @return INLINE_SIZE for PK index.
+     */
+    public int getPrimaryKeyInlineSize() {
+        return pkInlineSize != null ? pkInlineSize : -1;
+    }
+
+    /**
+     * Sets INLINE_SIZE for PK index. Implemented at the child.
+     *
+     * @param pkInlineSize INLINE_SIZE for PK index.
+     * @return {@code this} for chaining.
+     */
+    public QueryEntity setPrimaryKeyInlineSize(int pkInlineSize) {
```

Comment:
I got critical failure when I set "-2" here. Also I don't like idea when user can explicitly set "-1". For me it's something like undefined behavior. Can we restrict it with only positive values only?

`[2021-12-06 11:52:17,602][ERROR][exchange-worker-#56%index.BasicIndexTest%][IgniteTestResources] Critical system error detected. Will be handled accordingly to configured handler [hnd=StopNodeFailureHandler [super=AbstractFailureHandler [ignoredFailureTypes=UnmodifiableSet [SYSTEM_WORKER_BLOCKED, SYSTEM_CRITICAL_OPERATION_TIMEOUT]]], failureCtx=FailureContext [type=SYSTEM_WORKER_TERMINATION, err=java.lang.AssertionError]]
`

Reply:
Great issue. Thanks!